README.md

```diff
@@ -159,11 +159,51 @@ It also supports the embedded `Not()` matcher; e.g.
 
 `Ω(err == nil).Should(Not(BeTrue()))` => `Ω(x).Should(HaveOccurred())`
 
+### Wrong Comparison Assertion
+The linter finds assertion of boolean comparisons, which are already supported by existing gomega matchers. 
+
+The linter assumes that when compared something to literals or constants, these values should be used for the assertion,
+and it will do its best to suggest the right assertion expression accordingly. 
+
+There are several wrong patterns:
+```go
+var x = 10
+var s = "abcd"
+
+...
+
+Expect(x == 10).Should(BeTrue()) // should be Expect(x).Should(Equal(10))
+Expect(10 == x).Should(BeTrue()) // should be Expect(x).Should(Equal(10))
```

Comment:
What about `Expect(ptr == nil).Should(BeTrue())`?

Does it suggest `Expect(ptr).Should(Equal(nil))` or `Expect(ptr).Should(BeNil)`?

The former would trigger another warning when checked again, wouldn't it?

I don't know how deep down into the rabbit hole we want to go. It's better if the suggestion is final, but I also wouldn't make the code too complex to achieve that. Up to you...

Reply:
Thanks for your comment @pohly. This is implemented from (almost) day one, and no, the `Expect(ptr).Should(BeNil())` is a valid pattern and won't trigger another warning. 

We don't want `BeNil()` for errors, but for pointers it's ok.

Reply:
What I meant is that "former" == `Expect(ptr).Should(Equal(nil))` would (?) trigger another warning.

Reply:
No. nil and error comparisons are checked before the general one. No new triggers.

Reply:
Found a bug in the current implementation.

When comparing pointrs, go compares theier addresses. But the Equal matcher compares their values.